dts/bindings/stepper/gpio-stepper-controller.yaml

```diff
@@ -0,0 +1,30 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO Stepper Controller cluster for darlington transistor arrays or dual H-bridge
+
+compatible: "gpio-steppers"
+
+child-binding:
+  description: GPIO Controller for stepper motor
+  include:
+    - name: stepper-controller.yaml
+      property-allowlist:
+        - micro-step-res
+
+  properties:
+    gpios:
+      type: phandle-array
+      required: true
+      description: |
+        The gpio pin array on which the stepper inputs are to be connected
+        stepper {
+              compatible = "gpio-stepper";
+              status = "okay";
+              motor:motor {
+                gpios = <&gpioa 9 GPIO_ACTIVE_HIGH>,
+                        <&gpioc 7 GPIO_ACTIVE_HIGH>,
+                        <&gpiob 0 GPIO_ACTIVE_HIGH>,
+                        <&gpioa 7 GPIO_ACTIVE_HIGH>;
+              };
```

Comment:
why is this a child? please, just use one entry per stepper, this doesn't make sense, you're using one device per entry anyway, so there's no gain and an uncommon pattern in Zephyr

Reply:
You are right. I wanted to actually set this as an example for the drivers who can drive more than one steppers. I thought this would serve as an example, as to how those drivers could be shown in device tree.

Reply:
discussed offline, can be kept using `gpio-steppers`, will finish review later/tomorrow